### PR TITLE
#6970 Add more examples for utilizing custom cache type

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request

This pull request (PR) adds more examples for utilizing custom cache type

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.4/extension-dev-guide/cache/partial-caching/create-cache-type.html

## Links to Magento source code
-  src/guides/v2.4/extension-dev-guide/cache/partial-caching/create-cache-type.md

## Links to issue
- https://github.com/magento/devdocs/issues/6970

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 1]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
 *  `%cache_type_id%` defines the unique identifier of a cache type.
 *  `%CACHE_TYPE_TAG%` defines the unique tag to be used in the cache type scoping.
 
+## Store data into custom cache type {#m2devgde-store-data-cache-type}
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:84]:
```suggestion
## Store data in a custom cache type {#m2devgde-store-data-cache-type}
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 2]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+To store data in a serialized form into a custom cache should be made the steps:
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:86]:
```suggestion
To store data in a serialized form in a custom cache, follow these steps:
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 3]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+*  Pass the argument to the constructor `Magento\Framework\App\CacheInterface` $cache of a necessary class (Repository, Model, Block, etc)
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:88]:
```suggestion
*  Pass the argument to the constructor `Magento\Framework\App\CacheInterface` `$cache` of a required class (Repository, Model, Block, etc)
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 4]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+$cacheTag  = \VendorName\ModuleName\Model\Cache\Type\CacheType::CACHE_TAG;
+
+$storeData = $this->cache->save(
+    $this->serializer->serialize($cacheData),
+    $cacheKey,
+    [$cacheTag],
+    86400
+);
+```
+## Retrieve data from custom cache type {#m2devgde-retrieve-data-cache-type}
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:115]: Add a blank line before the header

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 5]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+Retrieve data in the cache
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:117]:
```suggestion
Retrieve data from the cache
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 6]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+```php
+$cacheKey  = \VendorName\ModuleName\Model\Cache\Type\CacheType::TYPE_IDENTIFIER;
+
+$data = $this->serializer->unserialize($this->cache->load($cacheKey));
+```
+
+## Invalidate custom cache type {#m2devgde-invalidate-cache-type}
+
+To invalidate custom cache type should be made the steps:
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:127]:
```suggestion
To invalidate a custom cache type, follow these steps:
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 7]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+*  Pass the argument to the constructor `Magento\Framework\App\Cache\TypeListInterface` $typeList of a necessary class (Repository, Model, Block, etc)
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:129]:
```suggestion
*  Pass the argument to the constructor `Magento\Framework\App\Cache\TypeListInterface` `$typeList` of a required class (Repository, Model, Block, etc)
```

File: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md [hunk 8]
```diff
@@ -81,5 +81,83 @@ You must specify the following parameters:
...
+
+$this->typeList->invalidate($cacheKey);
+```
+
+## Flush custom cache type {#m2devgde-flush-cache-type}
+
+The custom cache type can be flushed in the following ways:
+
+*  Go to System -> Cache Management and flush the custom cache type
+*  Programatically, using the TypeList.
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md:154]:
```suggestion
*  Programmatically, using the TypeList.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice additions - just some grammar corrections

**Comment by DenisSaltanahmedov** [comment]
@BarnyShergold thanks for your help. I have fixed the review.

**Review by BarnyShergold — APPROVED** [review]
Nice work!

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**